requirements.txt

```diff
@@ -4,7 +4,8 @@ cirq==0.10.0.dev20201203012958
 
 # third-party integration
 pyquil~=2.18.0
-qiskit~=0.16.2
+qiskit-aer>0.7
+qiskit-ibmq-provider>0.11
```

Comment:
Where is this required?

Reply:
Found it! Without ibmq-provider a runtime warning is raised, but tests still pass:

```text
Document: guide/guide-qaoa-landscape
------------------------------------
.../python3.6/site-packages/qiskit/__init__.py:75: RuntimeWarning: Could not import the IBMQ provider from the qiskit-ibmq-provider package. Install qiskit-ibmq-provider or check your installation.
  RuntimeWarning)

```

Reply:
I managed to get a working version in #487 which only installs qiskit-aer and qiskit-terra. I'm actually not sure why [guide-ibmq-backends](https://github.com/unitaryfund/mitiq/blob/master/docs/source/guide/guide-ibmq-backends.rst) passes though because this uses the ibmq provider.

Reply:
So, I guess the dependency resolver is trying to resolve the dependency between `aer` and `ibmq-provider` when it downloads all the versions of `terra`. 

Also, I think ` guide/guide-ibmq-backends` passes because there is no test being run for it. The next step skips to doctest summary. https://github.com/unitaryfund/mitiq/runs/1644604484?check_suite_focus=true#step:5:283

Reply:
Thanks @purva-thakre. Its unfortunate the dependency resolver is failing here. To me, it's not a big deal to just make qiskit the requirement in light of this. This is more favorable than adding more complexity with constraints, etc. So, unless there is a simple solution (and you're willing to keep searching!) I would vote for keeping qiskit as the requirement at this point.

Reply:
I have been thinking of putting this on hold until a better version of dependency resolver is released. I tried specifying the dependencies exactly via `==` but the dependency resolver was still backtracking across all versions of `terra`. I don't think specifying the constraints in `constraint.txt` file will work since my plan was to use `==` when pip installed via `requirements.txt`.

Reply:
This sounds good to me.